sdk/src/wallet/account/operations/balance.rs

```diff
@@ -89,178 +92,112 @@ where
                         }
                         Output::Foundry(output) => {
-                            // Add amount
-                            balance.base_coin.total += output.amount();
-                            // Add storage deposit
-                            balance.required_storage_deposit.foundry += rent;
-                            if !account_details.locked_outputs.contains(output_id) {
+                            output_balance.required_storage_deposit.foundry += rent;
+                            output_balance.foundries.push(output.id());
+                        }
+                        Output::Nft(output) => {
+                            output_balance.required_storage_deposit.nft += rent;
+                            output_balance.nfts.push(output.nft_id_non_null(output_id));
+                        }
+                        _ => {}
+                    }
+
+                    if !account_details.locked_outputs.contains(output_id) {
+                        if output.is_basic() {
+                            // Amount for basic outputs isn't added to total_rent_amount if there aren't native tokens,
+                            // since we can spend it without burning.
+                            if output
+                                .native_tokens()
+                                .map(|native_tokens| !native_tokens.is_empty())
+                                .unwrap_or(false)
+                            {
                                 total_rent_amount += rent;
                             }
-                            // Add native tokens
-                            total_native_tokens.add_native_tokens(output.native_tokens().clone())?;
-
-                            balance.foundries.push(output.id());
+                        } else {
+                            total_rent_amount += rent;
                         }
-                        _ => {
-                            // If there is only an [AddressUnlockCondition], then we can spend the output at any time
-                            // without restrictions
-                            if let [UnlockCondition::Address(_)] = output
-                                .unlock_conditions()
-                                .expect("output needs to have unlock conditions")
-                                .as_ref()
-                            {
-                                // add nft_id for nft outputs
-                                if let Output::Nft(output) = &output {
-                                    let nft_id = output.nft_id_non_null(output_id);
-                                    balance.nfts.push(nft_id);
-                                }
+                    }
+
+                    if let Some(native_tokens) = output.native_tokens() {
+                        total_native_tokens.add_native_tokens(native_tokens.clone())?;
+                    }
```

Comment:
So where do you check if we can actually unlock these?

Reply:
Hmm yeah you're right. I wanted to split more changes to later but it won't work without it.